app/src/androidTest/java/org/mozilla/focus/activity/VPNIconTest.java

```diff
@@ -0,0 +1,85 @@
+package org.mozilla.focus.activity;
+
+import android.content.Intent;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mozilla.focus.R;
+import org.mozilla.focus.helper.BeforeTestTask;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.doesNotExist;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static org.hamcrest.CoreMatchers.allOf;
+
+@RunWith(AndroidJUnit4.class)
+public class VPNIconTest {
+
+    @Rule
+    public final ActivityTestRule<MainActivity> activityRule = new ActivityTestRule<>(MainActivity.class, true, false);
+
+    @Before
+    public void setUp() {
+        new BeforeTestTask.Builder()
+                .setSkipFirstRun(true)
+                .build()
+                .execute();
+        activityRule.launchActivity(new Intent());
+    }
+
+    /**
+     * Test case no: TC0168
+     * Test case name: Show VPN icon
+     * Steps:
+     * 1. Launch app
+     * 2. VPN icon displayed
+     * 3. Relaunch app
+     * 4. VPN icon displayed
+     */
+    @Test
+    public void showVpnIcon() {
+
+        // VPN icon displayed
+        onView(withId(R.id.home_wifi_vpn_survey)).check(matches(isDisplayed()));
+
+        // Relaunch app
+        relaunchActivity();
+
+        // VPN icon displayed
+        onView(withId(R.id.home_wifi_vpn_survey)).check(matches(isDisplayed()));
+    }
+
+    /**
+     * Test case no: TC0171
+     * Test case name: Deny to install VPN
+     * Steps:
+     * 1. Launch app
+     * 2. Tap VPN icon
+     * 3. Tap No
+     * 4. Check VPN icon not exist
+     */
+    @Test
+    public void denyToInstallVpn() {
+
+        // VPN icon displayed
+        onView(allOf(withId(R.id.home_wifi_vpn_survey), isDisplayed())).perform(click());
+
+        // Deny to install
+        onView(allOf(withId(R.id.wifi_vpn_btn_no), isDisplayed())).perform(click());
+
+        // Check VPN icon not exist
+        onView(withId(R.id.wifi_vpn_btn_no)).check(doesNotExist());
+    }
+
+    private void relaunchActivity() {
+        activityRule.finishActivity();
```

Comment:
do you want to recreate the app?
This will only make the app goes to Recent Apps and show it again.
not sure if 
`activityRule.runOnUiThread { activityRule.activity.recreate() }`
will give you what you want.